backend/src/main/java/com/cic/service/MatchingService.java

```diff
@@ -10,18 +10,34 @@
 public class MatchingService {
 
   private final MatchingAlgorithm matcher;
+  private final float acceptedThreshold;
 
-  public MatchingService(MatchingAlgorithm matcher) {
+  public MatchingService(final MatchingAlgorithm matcher, final float acceptedThreshold) {
     this.matcher = matcher;
+    this.acceptedThreshold = acceptedThreshold;
   }
 
-  public List<Match> returnMatches(Set<Person> mentees, Set<Person> mentors) {
-    List<Match> matches = new ArrayList<>();
-
-    // for (int i = 0; i < mentees.length; i++) {
-    // matches.add(new Match().mentor(mentees[i]).mentee(mentors[i]));
-    // }
-
+  public List<Match> returnMatches(final Set<Person> mentees, final Set<Person> mentors) {
+    final List<Match> matches = new ArrayList<>();
+    final Set<Person> tempMentors = mentors;
+    mentees.forEach(mentee -> {
+      Person mentor = findGoodMatch(mentee, tempMentors);
+      Match match = new Match();
+      match.setMentee(mentee);
+      match.setMentor(mentor);
+      matches.add(match);
+      tempMentors.remove(mentor);
+    });
     return matches;
   }
+
+  private Person findGoodMatch(Person mentee, Set<Person> mentors) {

```

Comment:
Would a name like `findMentorForMentee` be a better name? Since it isn't clear to me what the function is finding a match for from the name of the function

Reply:
done